### PR TITLE
Mining Rework 1.5 - Okay now lets blow things up!

### PR DESCRIPTION
**Pull request by ToasterBiome**
# Document the changes in your pull request

**Multiple changes to the plasma magmite/lavaland cycle.**

1. The **world anvil** is no longer located at the necropolis gate. Instead, it's in it's own 0 cost, always spawning ruin. It has a GPS signal to aid you in the search, "Tempered Signal".
image
2. The way the **world anvil** functions is different. You heat it the same way, and then use a piece of magmite on it. After 10 seconds, it will turn into **plasma magmite upgrade parts**. 
image
Use these on a valid PKA/plasma cutter and it will turn them into their mega versions. Beware, as they go inert in 10 minutes, losing their glow and no longer being useful. You can restore this by returning to the **world anvil** and using them on the anvil again.
3. Adds two new **mining charges**, and gives them new sprites from @UselessTheremin, a **lesser mining charge** and a **mega mining charge**. Here's all 3 of them side by side:
image
regular (green) mining charges stay the same, while lesser charges get a far smaller radius, and mega gets a bigger one.
4. Regular mining charges are removed from the mining equipment vendor. They are replaced with **lesser mining charges** for 300 mining points instead of 500.
5. You may buy a duffelbag of 3 **lesser mining charges** with your mining voucher to help you get started.
6. There is a new research: **Magmite Mining**. It requires advanced mining, and deconstruction of a piece of **plasma magmite**, along with 1200 research points.
7. (Regular) **mining charges** are moved to a protolathe item requiring Advanced Mining, costing 3000u iron, and 1000u plasma.
8. **Mega mining charges** are a new protolathe item that requires the Magmite Mining research. They are the biggest mining charge available and cost 5000 iron, 2000 plasma, 1500 uranium, and 500 diamond.
9. Adds the **mega plasma cutter shotgun**, which you can get by upgrading a plasma cutter shotgun. It features longer range, and auto defuses Gibtonite without needing to upgrade.
image
10. Also moves mining charges from grenades.dmi to mining.dmi, along with moving magmite out of the ores_coins.dm and into a new code/modules/mining/equipment/upgrades.dm along with the new magmite parts. This is not player facing.

# Wiki Documentation

All of this. 

# Changelog

:cl:  ToasterBiome, LazennG, Chubbygummibear, UselessTheremin, tatax, cuackles, stwordcitizen
tweak: moves world anvil to it's own always spawning ruin in lavaland
tweak: reworks how the world anvil functions
rscadd: adds magmite upgrade parts and moves upgrading functionality to it
rscadd: adds mega plasma shotgun
rscadd: adds two new mining charges
rscadd: adds new mining research based around magmite
rscadd: adds bag of lesser mining charges for voucher
experimental: shuffles around when you get mining charges/which kind
/:cl:

### CODE REVIEW

File: code/modules/projectiles/guns/energy/special.dm [hunk 1]
```diff
@@ -231,17 +231,26 @@
 	name = "plasma cutter shotgun"
 	icon_state = "miningshotgun"
 	item_state = "miningshotgun"
-	desc = "An industrial-grade heavy-duty mining shotgun"
+	desc = "An industrial-grade heavy-duty mining shotgun."
```
Comment (Anvilman6) [inline code/modules/projectiles/guns/energy/special.dm]:
```suggestion
	desc = "An industrial-grade, heavy-duty mining shotgun."
```

File: code/modules/projectiles/guns/energy/special.dm [hunk 2]
```diff
@@ -231,17 +231,26 @@
...
 	if(try_upgrade(I))
 		to_chat(user, span_notice("You install [I] into [src]"))
 		playsound(loc, 'sound/items/screwdriver.ogg', 100, 1)
 		qdel(I)
 
+/obj/item/gun/energy/plasmacutter/scatter/mega
+	name = "mega plasma cutter shotgun"
+	icon_state = "miningshotgun_mega"
+	item_state = "miningshotgun_mega"
+	desc = "An industrial-grade heavy-duty mining shotgun. This one seems... mega!"
```
Comment (Anvilman6) [inline code/modules/projectiles/guns/energy/special.dm]:
```suggestion
	desc = "An industrial-grade, heavy-duty mining shotgun. This one seems... mega!"
```

File: code/modules/mining/lavaland/world_anvil.dm [hunk 3]
```diff
@@ -31,76 +45,29 @@
...
-			RegisterSignal(I, COMSIG_MOVABLE_MOVED, .proc/ItemMoved,TRUE)
-
-/obj/structure/world_anvil/proc/ItemMoved(obj/item/I, atom/OldLoc, Dir, Forced)
-	vis_contents -= I
-	placed_objects -= I
-	UnregisterSignal(I, COMSIG_MOVABLE_MOVED)
+		return
+	if(istype(I,/obj/item/magmite))
+		if(forge_charges <= 0)
+			to_chat(user,"The anvil is not heated enough to be usable!")
```
Comment (Anvilman6) [inline code/modules/mining/lavaland/world_anvil.dm]:
```suggestion
			to_chat(user,"The anvil is not hot enough to be usable!")
```

File: code/modules/mining/lavaland/world_anvil.dm [hunk 4]
```diff
@@ -31,76 +45,29 @@
...
+			return
+		if(do_after(user,10 SECONDS, target = src))
+			new /obj/item/magmite_parts(src)
+			qdel(I)
+			to_chat(user, "You forge the plasma magmite into plasma magmite upgrade parts.")
```
Comment (Anvilman6) [inline code/modules/mining/lavaland/world_anvil.dm]:
```suggestion
			to_chat(user, "You carefully forge the rough plasma magmite into plasma magmite upgrade parts.")
```

File: code/modules/mining/lavaland/world_anvil.dm [hunk 5]
```diff
@@ -31,76 +45,29 @@
...
+			if(!forge_charges)
+				visible_message("The world anvil cools down.")
+	if(istype(I,/obj/item/magmite_parts))
+		if(forge_charges <= 0)
+			to_chat(user,"The anvil is not heated enough to be usable!")
```
Comment (Anvilman6) [inline code/modules/mining/lavaland/world_anvil.dm]:
```suggestion
			to_chat(user,"The anvil is not hot enough to be usable!")
```

File: code/modules/mining/lavaland/world_anvil.dm [hunk 6]
```diff
@@ -31,76 +45,29 @@
...
+			return
+		var/obj/item/magmite_parts/parts = I
+		if(!parts.inert)
+			to_chat(user,"The magmite upgrade parts are already glowing!")
```
Comment (Anvilman6) [inline code/modules/mining/lavaland/world_anvil.dm]:
```suggestion
			to_chat(user,"The magmite upgrade parts are already glowing and usable!")
```

Comment (Anvilman6) [reply code/modules/mining/lavaland/world_anvil.dm]: Synchronize across the PR if you want to call these parts "plasma magmite" or just "magmite"

File: code/modules/mining/lavaland/world_anvil.dm [hunk 7]
```diff
@@ -31,76 +45,29 @@
...
+			return
+		if(do_after(user,5 SECONDS, target = src))
+			parts.restore()
+			to_chat(user, "You restore the magmite upgrade parts' magma-like glow. They are now usable again.")
```
Comment (Anvilman6) [inline code/modules/mining/lavaland/world_anvil.dm]:
```suggestion
			to_chat(user, "You successfully reheat the magmite upgrade parts. They are now glowing and usable again.")
```

File: code/modules/mining/equipment/upgrades.dm [hunk 8]
```diff
@@ -0,0 +1,68 @@
+//plasma magmite is exclusively used to upgrade mining equipment, by using it on a heated world anvil to make upgradeparts.
+/obj/item/magmite
+	name = "plasma magmite"
+	desc = "A chunk of plasma magmite, crystallized deep under lavaland's surface. Its strength seems to fluctuate depending on the distance to the planet."
```
Comment (Anvilman6) [inline code/modules/mining/equipment/upgrades.dm]:
```suggestion
	desc = "A chunk of plasma magmite, crystallized deep under the planet's surface. It seems to lose strength as it gets further from the planet!"
```

File: code/modules/mining/equipment/upgrades.dm [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+	icon = 'icons/obj/mining.dmi'
+	icon_state = "Magmite ore"
+	w_class = WEIGHT_CLASS_NORMAL
+
+/obj/item/magmite_parts
+	name = "plasma magmite upgrade parts"
+	desc = "Forged on the lavaland anvil, these parts can be used to upgrade many kinds of mining equipment."
```
Comment (Anvilman6) [inline code/modules/mining/equipment/upgrades.dm]:
```suggestion
	desc = "Forged on the legendary World Anvil, these parts can be used to upgrade many kinds of mining equipment."
```

File: code/modules/mining/equipment/upgrades.dm [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+	addtimer(CALLBACK(src, .proc/go_inert), 10 MINUTES)
+
+/obj/item/magmite_parts/proc/go_inert()
+	if(inert)
+		return
+	visible_message(span_warning("The [src] loses it's glow!"))
+	inert = TRUE
+	name = "inert plasma magmite upgrade parts"
+	icon_state = "upgrade_parts_inert"
+	desc += " This one has lost it's magma-like glow."
```
Comment (Anvilman6) [inline code/modules/mining/equipment/upgrades.dm]:
```suggestion
	desc += "It appears to have lost its magma-like glow."
```

File: code/modules/mining/equipment/upgrades.dm [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+		return
+	inert = FALSE
+	name = initial(name)
+	icon_state = initial(icon_state)
+	desc = initial(desc)
+	addtimer(CALLBACK(src, .proc/go_inert), 10 MINUTES)
+
+/obj/item/magmite_parts/afterattack(atom/target, mob/user, proximity_flag, click_parameters)
+	if(inert)
+		to_chat(span_warning("[src] is inert! Restore its glow at the world anvil!"))
```
Comment (Anvilman6) [inline code/modules/mining/equipment/upgrades.dm]:
```suggestion
		to_chat(span_warning("[src] appears inert! Perhaps the World Anvil can restore it!"))
```
Mentions of the World Anvil should probably be capitalized throughout, as it's a singular, important entity. Should also include calls to "anvil" so long as it's referencing the World Anvil specifically.

File: code/modules/mining/equipment/mining_charges.dm [hunk 12]
```diff
@@ -43,3 +44,19 @@
...
+
+/obj/item/grenade/plastic/miningcharge/lesser
+	name = "lesser mining charge"
+	desc = "A lesser version of our boys in green. Used to make big holes. Only works on rocks!"
+	icon_state = "mining-charge-1"
+	boom_sizes = list(1,1,1)
+
+/obj/item/grenade/plastic/miningcharge/mega
+	name = "mega mining charge"
+	desc = "A lesser version of our boys in green. Used to make big holes. Only works on rocks!"
```
Comment (Anvilman6) [inline code/modules/mining/equipment/mining_charges.dm]:
```suggestion
	desc = "A mining charge. This one seems much more powerful than normal!"
```

File: code/modules/mining/equipment/mining_charges.dm [hunk 13]
```diff
@@ -43,3 +44,19 @@
 /obj/item/grenade/plastic/miningcharge/deconstruct(disassembled = TRUE) //no gibbing a miner with pda bombs
 	if(!QDELETED(src))
 		qdel(src)
+
+/obj/item/grenade/plastic/miningcharge/lesser
+	name = "lesser mining charge"
+	desc = "A lesser version of our boys in green. Used to make big holes. Only works on rocks!"
```
Comment (Anvilman6) [inline code/modules/mining/equipment/mining_charges.dm]:
```suggestion
	desc = "A mining charge. This one seems less powerful than normal. Only works on rocks!"
```

File: code/modules/mining/equipment/mining_charges.dm (hunk 12 above)
Comment (Anvilman6) [reply code/modules/mining/equipment/mining_charges.dm]: Assuming this wasn't supposed to be a copy/paste of the above. Rock clause may need re-adding if that's still the case

File: code/datums/ruins/lavaland.dm [hunk 14]
```diff
@@ -230,6 +230,15 @@
 	suffix = "lavaland_surface_puzzle.dmm"
 	cost = 5
 
+/datum/map_template/ruin/lavaland/worldanvil //Plasma magmite upgrading area... always place.
+	name = "World Anvil"
+	id = "worldanvil"
+	description = "An ancient anvil stands untained for millenia."
```
Comment (Anvilman6) [inline code/datums/ruins/lavaland.dm]:
```suggestion
	description = "An ancient anvil, standing untainted for millennia. Wonders were once forged here."
```

**Review by Anvilman6 — CHANGES_REQUESTED** [review]
`My main goal is to blow up and then act like I don't know nobody`

Surely putting more explosives in the hands of miners would never go wrong.

File: code/modules/mining/equipment/upgrades.dm [hunk 15]
```diff
@@ -0,0 +1,68 @@
...
+		qdel(gun)
+		var/obj/item/gun/energy/plasmacutter/adv/mega/newgun = new(src)
+		user.put_in_hand(newgun)
+		to_chat(user,"Harsh tendrils wrap around the plasma cutter, merging the parts and cutter to form a mega plasma cutter.")
+	if(target.type == /obj/item/gun/energy/plasmacutter/scatter) //holy fuck make a new system bro do a /datum/worldanvilrecipe DAMN
+		var/obj/item/gun/energy/plasmacutter/scatter/gun = target
+		qdel(gun)
+		var/obj/item/gun/energy/plasmacutter/scatter/mega/newgun = new(src)
+		user.put_in_hand(newgun)
+		to_chat(user,"Harsh tendrils wrap around the plasma cutter shotgun, merging the parts and cutter to form a mega plasma cutter shotgun.")
```
Comment (SomeguyManperson) [inline code/modules/mining/equipment/upgrades.dm]: image

File: code/modules/mining/equipment/mining_charges.dm (hunk 13 above)
Comment (ToasterBiome) [reply code/modules/mining/equipment/mining_charges.dm]: but i really like this description :(

File: code/modules/mining/equipment/upgrades.dm (hunk 15 above)
Comment (ToasterBiome) [reply code/modules/mining/equipment/upgrades.dm]: Fuck you

File: code/modules/mining/lavaland/world_anvil.dm [hunk 16]
```diff
@@ -31,76 +45,29 @@
...
-			RegisterSignal(I, COMSIG_MOVABLE_MOVED, .proc/ItemMoved,TRUE)
-
-/obj/structure/world_anvil/proc/ItemMoved(obj/item/I, atom/OldLoc, Dir, Forced)
-	vis_contents -= I
-	placed_objects -= I
-	UnregisterSignal(I, COMSIG_MOVABLE_MOVED)
+		return
+	if(istype(I,/obj/item/magmite))
+		if(forge_charges <= 0)
+			to_chat(user,"The anvil is not hot enough to be usable!")
```
Comment (Anvilman6) [inline code/modules/mining/lavaland/world_anvil.dm]:
```suggestion
			to_chat(user,"The World Anvil is not hot enough to be usable!")
```

File: code/modules/mining/lavaland/world_anvil.dm [hunk 17]
```diff
@@ -31,76 +45,29 @@
...
+			return
+		if(do_after(user,10 SECONDS, target = src))
+			new /obj/item/magmite_parts(src)
+			qdel(I)
+			to_chat(user, "You carefully forge the rough plasma magmite into plasma magmite upgrade parts.")
+			if(!forge_charges)
+				visible_message("The world anvil cools down.")
+	if(istype(I,/obj/item/magmite_parts))
+		if(forge_charges <= 0)
+			to_chat(user,"The anvil is not hot enough to be usable!")
```
Comment (Anvilman6) [inline code/modules/mining/lavaland/world_anvil.dm]:
```suggestion
			to_chat(user,"The World Anvil is not hot enough to be usable!")
```

File: code/modules/mining/lavaland/world_anvil.dm [hunk 18]
```diff
@@ -31,76 +45,29 @@
...
+	if(istype(I,/obj/item/magmite))
+		if(forge_charges <= 0)
+			to_chat(user,"The anvil is not hot enough to be usable!")
+			return
+		if(do_after(user,10 SECONDS, target = src))
+			new /obj/item/magmite_parts(src)
+			qdel(I)
+			to_chat(user, "You carefully forge the rough plasma magmite into plasma magmite upgrade parts.")
+			if(!forge_charges)
+				visible_message("The world anvil cools down.")
```
Comment (Anvilman6) [inline code/modules/mining/lavaland/world_anvil.dm]:
```suggestion
				visible_message("The World Anvil cools down.")
```

File: code/modules/mining/lavaland/world_anvil.dm [hunk 19]
```diff
@@ -31,76 +45,29 @@
...
+			return
+		var/obj/item/magmite_parts/parts = I
+		if(!parts.inert)
+			to_chat(user,"The magmite upgrade parts are already glowing and usable!")
+			return
+		if(do_after(user,5 SECONDS, target = src))
+			parts.restore()
+			to_chat(user, "You successfully reheat the magmite upgrade parts. They are now glowing and usable again.")
+			if(!forge_charges)
+				visible_message("The world anvil cools down.")
```
Comment (Anvilman6) [inline code/modules/mining/lavaland/world_anvil.dm]:
```suggestion
				visible_message("The World Anvil cools down.")
```

File: code/modules/mining/lavaland/world_anvil.dm [hunk 20]
```diff
@@ -31,76 +45,29 @@
 		to_chat(user,"You place down the gibtonite on the world anvil, and watch as the gibtonite melts into it. The world anvil is now heated enough for [forge_charges] forge[forge_charges > 1 ? "s" : ""].")
```
Comment (Anvilman6) [inline code/modules/mining/lavaland/world_anvil.dm]: If we're going to capitalize instances of "World Anvil", we'll need to be comprehensive. Check upwards through the other references in this file too.

Comment (Anvilman6) [reply code/modules/mining/lavaland/world_anvil.dm]:
```suggestion
		to_chat(user,"You place down the gibtonite on the World Anvil, and watch as the gibtonite melts into it. The World Anvil is now heated enough for [forge_charges] forge[forge_charges > 1 ? "s" : ""].")
```

**Review by Anvilman6 — CHANGES_REQUESTED** [review]
Back by unpopular demand: me

**Review by Anvilman6 — APPROVED** [review]
Mine

Good luck lavaland

**Comment by JamieD1** [comment]
Were you supposed to remove the empty icon?
Also @grimm-hollowknight

**Comment by ToasterBiome** [comment]
> Were you supposed to remove the empty icon? Also @grimm-hollowknight

There is no empty icon

**Comment by JamieD1** [comment]
Comments in Discord

**Comment by KillerOrcaCora** [comment]
Sprites look fine.

**Review by KillerOrcaCora — APPROVED** [review]
Everything looks alright sprite wise.

**Review by KillerOrcaCora — APPROVED** [review]

**Review by JamieD1 — APPROVED** [review]

**Pull request merged**